CONTRIBUTING.md

```diff
@@ -1,7 +1,119 @@
+    1. [Setup a conjur OSS Environment](#setup-a-conjur-oss-environment)
+    2. [Load policy to set up Conjur Ansible integration](#load-policy-to-set-up-conjur-ansible-integration)
+    3. [Create Ansible managed nodes](#create-ansible-managed-nodes)
+    4. [Use `conjur_host_identity` to set up Conjur identity on managed nodes](#use-conjur-host-identity-to-set-up-conjur-identity-on-managed-nodes)
+    5. [Use `conjur_variable` lookup plugin to provide secrets to Ansible Playbooks](#use-conjur-variable-lookup-plugin-to-provide-secrets-to-ansible-playbooks)
+
+
+ ## Prerequisites
+
+To start developing and testing using our development scripts ,
+the following tools need to be installed:
+
+1. [Git][get-git] to manage source code
+2. [Docker][get-docker] to manage dependencies and runtime environments
+3. [Docker Compose][get-docker-compose] to orchestrate Docker environments
+
+[get-docker]: https://docs.docker.com/engine/installation
+[get-docker-compose]: https://docs.docker.com/compose/install
+[get-git]: https://git-scm.com/downloads
+
+## Set up a development environment
+
+The `dev` directory contains a `docker-compose` file which creates a development
+environment :
+-  A Conjur Open Source instance
+-  An Ansible control node
+-  Managed nodes to push tasks to
+
+To use it:
+
+1. Install dependencies (as above)
+
+1. To setup the dev environment ,first need to Clone GitHub [conjur-collection](https://github.com/cyberark/ansible-conjur-collection) repository in your directory and then run start_oss.sh script
+
+
+ ```sh-session
+ $ git clone https://github.com/cyberark/ansible-conjur-collection.git
+ $ cd ansible-conjur-collection/dev
+ $ ./start.sh
+
+ ```
+### Verification
+
+  When start_oss.sh script successfully setup Conjur environment along with inventory machines , the terminal returns the following:
+
+   ```sh-session
+   ...
+   PLAY RECAP *********************************************************************
+   ansibleplugingtestingconjurhostidentity-test_app_centos-1 : ok=17 ...
+   ansibleplugingtestingconjurhostidentity-test_app_centos-2 : ok=17 ...
+   ansibleplugingtestingconjurhostidentity-test_app_ubuntu-1 : ok=16 ...
+   ansibleplugingtestingconjurhostidentity-test_app_ubuntu-2 : ok=16 ...
+
+   ```
+
+   After starting Conjur, your instance will be configured with the following:
+   * Account: `cucumber`
+   * User: `admin`
+   * Password: Run `conjurctl role retrieve-key cucumber:user:admin` inside the Conjur container shell to retrieve the admin user API key (which is also the  password)
+
+### Useful links
+
+- [Official documentation for Conjur's Ansible integration](https://docs.conjur.org/Latest/en/Content/Integrations/ansible.html)
+- [Conjur Collection on Ansible Galaxy](https://galaxy.ansible.com/cyberark/conjur)
+- [Ansible documentation for the Conjur collection](https://docs.ansible.com/ansible/latest/collections/cyberark/conjur/index.html)
+
+## Testing Conjur OSS
```

Comment:
We should change the `Testing Conjur OSS`, `Testing Conjur Enterprise`, and `Generate conjur variable unit tests report` sections into these:

```
## Testing

### Unit tests

Unit tests are only available for the Conjur Variable Lookup plugin. To run these tests:

\```
./dev/test_unit.sh
\```

### Integration tests

The collection has integration tests for both the Variable Lookup plugin and the Host Identity role that will validate each against live Conjur and Ansible containers.

To run all tests:
\```
./ci/test.sh -a
\```

To run the tests for a particular module:
\```
./ci/test.sh -d <role or plugin name>
\```

Integration tests can be run against Conjur Enterprise by adding the `-e` flag:
\```
./ci/test/sh -e -a
\```
```